VRUI.java

```diff
@@ -0,0 +1,224 @@
+		for ( Customer customer: customers ) {
+			if ( customer.getName().equals(customerName)) {
+				foundCustomer = customer ;
+				break ;
+			}
+		}
+		if ( foundCustomer == null ) return ;
+		
+		System.out.println("Enter video title to return: ") ;
+		String videoTitle = scanner.next() ;
+			
+		List<Rental> customerRentals = foundCustomer.getRentals() ;
+		for ( Rental rental: customerRentals ) {
+			if ( rental.getVideo().getTitle().equals(videoTitle) && rental.getVideo().isRented() ) {
+				rental.returnVideo();
+				rental.getVideo().setRented(false);
+				break ;
+			}
+		}		
+	}
+
+	private void init() {
+		Customer james = new Customer("James") ;
+		Customer brown = new Customer("Brown") ;
+		customers.add(james) ;
+		customers.add(brown) ;
+		
+		Video v1 = new Video("v1", Video.CD, Video.REGULAR, new Date()) ;
+		Video v2 = new Video("v2", Video.DVD, Video.NEW_RELEASE, new Date()) ;
+		videos.add(v1) ;
+		videos.add(v2) ;
+		
+		Rental r1 = new Rental(v1) ;
+		Rental r2 = new Rental(v2) ;
+		
+		james.addRental(r1) ;
+		james.addRental(r2) ;
+	}
+
+	public void listVideos() {
+		System.out.println("List of videos");
+		
+		for ( Video video: videos ) {
+			System.out.println("Price code: " + video.getPriceCode() +"\tTitle: " + video.getTitle()) ;
+		}
+		System.out.println("End of list");
+	}
+
+	public void listCustomers() {
+		System.out.println("List of customers");
+		for ( Customer customer: customers ) {
+			System.out.println("Name: " + customer.getName() +
+					"\tRentals: " + customer.getRentals().size()) ;
+			for ( Rental rental: customer.getRentals() ) {
+				System.out.print("\tTitle: " + rental.getVideo().getTitle() + " ") ;
+				System.out.print("\tPrice Code: " + rental.getVideo().getPriceCode()) ;
+			}
+		}
+		System.out.println("End of list");
+	}
+
+	public void getCustomerReport() {
+		System.out.println("Enter customer name: ") ;
+		String customerName = scanner.next() ;
+		
+		Customer foundCustomer = null ;
+		for ( Customer customer: customers ) {
+			if ( customer.getName().equals(customerName)) {
+				foundCustomer = customer ;
+				break ;
+			}
+		}
+
+		if ( foundCustomer == null ) {
+			System.out.println("No customer found") ;
+		} else {
+			String result = foundCustomer.getReport() ;
+			System.out.println(result);
+		}
+	}
+
+	public void rentVideo() {
+		System.out.println("Enter customer name: ") ;
+		String customerName = scanner.next() ;
+		
+		Customer foundCustomer = null ;
+		for ( Customer customer: customers ) {
+			if ( customer.getName().equals(customerName)) {
+				foundCustomer = customer ;
+				break ;
+			}
+		}
+
+		if ( foundCustomer == null ) return ;
+		
+		System.out.println("Enter video title to rent: ") ;
+		String videoTitle = scanner.next() ;
+
+		Video foundVideo = null ;
+		for ( Video video: videos ) {
+			if ( video.getTitle().equals(videoTitle) && video.isRented() == false ) {
+				foundVideo = video ;
+				break ;
+			}
+		}
+
+		if ( foundVideo == null ) return ;
```

Comment:
the code works perfect!
just need to add a `syso "No video on the list"` or something, inside the `if`